openc3/lib/openc3/api/cmd_api.rb

```diff
@@ -320,7 +320,20 @@ def cmd_implementation(method_name, *args, range_check:, hazardous_check:, raw:,
         'hazardous_check' => hazardous_check.to_s,
         'raw' => raw.to_s
       }
-      Logger.info(build_cmd_output_string(target_name, cmd_name, cmd_params, packet, raw), scope: scope) if !packet["messages_disabled"]
+      log_cmd = true
+      if packet["messages_disabled"]
+        log_cmd = false
+      else
+        cmd_params.each do |key, value|
+          item = packet['items'].find { |item| item['name'] == key.to_s }
+          if item['states'] && item['states'][value] && item['states'][value]["messages_disabled"]
+            log_cmd = false
+          end
+        end
```

Comment:
I don't think this is too process intensive but we do have to iterate through all the command parameters